### PR TITLE
discovery: refactor processNetworkAnnouncement into smaller functions

### PR DESCRIPTION
**Pull request by Crypt-iQ**
To improve the readability of the massive function. It may be possible to split them even further.

### CODE REVIEW

File: discovery/gossiper.go [hunk 1]
```diff
@@ -2984,3 +2752,232 @@ func (d *AuthenticatedGossiper) handleNodeAnnouncement(nMsg *networkMsg,
...
+
+// handleChanAnnouncement processes a new channel announcement.
+func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
+	ann *lnwire.ChannelAnnouncement,
+	ops []batch.SchedulerOption) ([]networkMsg, bool) {
+
+	// We'll ignore any channel announcements that target any chain other
+	// than the set of chains we know of.
+	if !bytes.Equal(ann.ChainHash[:], d.cfg.ChainHash[:]) {
+		err := fmt.Errorf("ignoring ChannelAnnouncement from chain=%v"+
```
Comment (positiveblue) [inline discovery/gossiper.go:2136]: nit: I think this fits in 80 characters

```
err := fmt.Errorf("ignoring ChannelAnnouncement from chain=%v, "+
    "gossiper on chain=%v", msg.ChainHash, d.cfg.ChainHash)
```

File: discovery/gossiper.go [hunk 2]
```diff
@@ -2984,3 +2752,232 @@ func (d *AuthenticatedGossiper) handleNodeAnnouncement(nMsg *networkMsg,
...
+		nMsg.err <- nil
+		return nil, true
+	}
+
+	// If this is a remote channel announcement, then we'll validate all
+	// the signatures within the proof as it should be well formed.
+	var proof *channeldb.ChannelAuthProof
+	if nMsg.isRemote {
+		if err := routing.ValidateChannelAnn(ann); err != nil {
+			err := fmt.Errorf("unable to validate announcement: "+
```
Comment (positiveblue) [inline discovery/gossiper.go:2177]: nit: fits in 80 characters

**Review by positiveblue — APPROVED** [review]
tACK! Thank you for breaking the review in those commits, really easy to follow :tada: 

nit: most of the logic was moved one indentation back, many lines can be reformatted and still meet the 80 columns constrain. Not blocking

File: discovery/gossiper.go (hunk 2 above)
Comment (Crypt-iQ) [reply discovery/gossiper.go:2177]: same here on my IDE it is at 80 characters

File: discovery/gossiper.go (hunk 1 above)
Comment (Crypt-iQ) [reply discovery/gossiper.go:2136]: on my IDE it is already 80 characters with tab size of 8, and this puts me at 82

File: discovery/gossiper.go [hunk 3]
```diff
@@ -2976,3 +2656,307 @@ func (d *AuthenticatedGossiper) handleChanUpdate(nMsg *networkMsg,
...
+	// proceeding further.
+	//
+	// We must acquire the mutex for this channel ID before getting the
+	// channel from the database, to ensure what we read does not change
+	// before we call AddProof() later.
+	d.channelMtx.Lock(ann.ShortChannelID.ToUint64())
+	defer d.channelMtx.Unlock(ann.ShortChannelID.ToUint64())
+
+	chanInfo, e1, e2, err := d.cfg.Router.GetChannelByID(
+		ann.ShortChannelID)
```
Comment (yyforyongyu) [inline discovery/gossiper.go:2703]: nit: add a new line here

**Review by yyforyongyu — COMMENTED** [review]
Something I wanted to do while reading the code, thank you for refactoring it! Changes looking good, only a few nits.

File: discovery/gossiper.go [hunk 4]
```diff
@@ -2981,3 +2673,306 @@ func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
...
+	// update announcement message. We'll need this to properly verify the
+	// message's signature.
+	//
+	// We make sure to obtain the mutex for this channel ID before we
+	// access the database. This ensures the state we read from the
+	// database has not changed between this point and when we call
+	// UpdateEdge() later.
+	d.channelMtx.Lock(upd.ShortChannelID.ToUint64())
+	defer d.channelMtx.Unlock(upd.ShortChannelID.ToUint64())
+	chanInf, e1, e2, err := d.cfg.Router.GetChannelByID(upd.ShortChannelID)
```
[the remark this answers is not in the document]
Comment (Crypt-iQ) [reply discovery/gossiper.go:2423]: 80 chars

File: discovery/gossiper.go [hunk 5]
```diff
@@ -2981,3 +2673,306 @@ func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
...
+	// quickly reject it.
+	timestamp := time.Unix(int64(upd.Timestamp), 0)
+	if d.cfg.Router.IsStaleEdgePolicy(
+		upd.ShortChannelID, timestamp, upd.ChannelFlags,
+	) {
+
+		log.Debugf("Ignored stale edge policy: peer=%v, source=%x, "+
+			"msg=%s, is_remote=%v", nMsg.peer,
+			nMsg.source.SerializeCompressed(), nMsg.msg.MsgType(),
+		)
```
[the remark this answers is not in the document]
Comment (Crypt-iQ) [reply discovery/gossiper.go:2407]: oops good catch

File: discovery/gossiper.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply discovery/gossiper.go:2423]: I think the benefits of having the same variable named the same way outweigh having everything in one line. Non-blocking tho, just a thought.

**Review by yyforyongyu — APPROVED** [review]
LGTM🎉

**Pull request merged**